shared/shared.go

```diff
@@ -75,26 +75,38 @@ func MakeLabelFunc() func(hash LabelHash, labelID uint64, leftSiblings [][]byte)
 	}
 }
 
+type Member struct {
+	Challenge []byte `scale:"max=32"`
+}
+
 type Proof struct {
 	// The actual proof.
 	MerkleProof
 
 	// Members is the ordered list of miners challenges which are included
 	// in the proof (by using the list hash digest as the proof generation input (the statement)).
-	Members [][]byte
+	Members []Member `scale:"max=150"`
 
 	// NumLeaves is the width of the proof-generation tree.
 	NumLeaves uint64
 }
 
 type ProofMessage struct {
 	Proof
-	ServicePubKey []byte
-	RoundID       string
+	ServicePubKey []byte `scale:"max=32"`
+	RoundID       string `scale:"max=32"`
+}
+
+type Leaf struct {
+	Value []byte `scale:"max=32"`
+}
+
+type Node struct {
+	Value []byte `scale:"max=32"`
 }
 
 type MerkleProof struct {
-	Root         []byte
-	ProvenLeaves [][]byte
-	ProofNodes   [][]byte
+	Root         []byte `scale:"max=32"`
+	ProvenLeaves []Leaf `scale:"max=150"`
```

Comment:
This number should be in-sync with the difficulty parameter: https://github.com/spacemeshos/poet/blob/53f681ea6d46d571d38afa8be9a4f77997a87afe/shared/shared.go#L16

IMO, it should be at least documented.